diff --git a/c++/src/kj/common.h b/c++/src/kj/common.h
index 6a3160999..4b5316163 100644
--- a/c++/src/kj/common.h
+++ b/c++/src/kj/common.h
@@ -109,10 +109,19 @@ KJ_BEGIN_HEADER
 #undef _GLIBCXX_HAVE_GETS
 #endif
 
-#if defined(_MSC_VER)
+#if _WIN32
+// Windows likes to define macros for min() and max(). We just can't deal with this.
+// If windows.h was included already, undef these.
+#undef min
+#undef max
+// If windows.h was not included yet, define the macro that prevents min() and max() from being
+// defined.
 #ifndef NOMINMAX
 #define NOMINMAX 1
 #endif
+#endif
+
+#if defined(_MSC_VER)
 #include <intrin.h>  // __popcnt
 #endif
 
diff --git a/c++/src/kj/windows-sanity.h b/c++/src/kj/windows-sanity.h
index 64475dc41..2e9ef6aa2 100644
--- a/c++/src/kj/windows-sanity.h
+++ b/c++/src/kj/windows-sanity.h
@@ -49,9 +49,8 @@
 #pragma once
 
 namespace win32 {
-  const auto ERROR_ = ERROR;
-
 #ifdef ERROR  // This could be absent if e.g. NOGDI was used.
+  const auto ERROR_ = ERROR;
 #undef ERROR
   const auto ERROR = ERROR_;
 #endif